### PR TITLE
Rework player controller

### PR DESCRIPTION
**Pull request by Yonimevicio**
Still fixing some bugs:
    - guns animations (fixed)
    - sounds (fixed)
    - dash (fixed)
 Still need to finish cleaning playercontroller.h (fixed)

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-mb5m9k Rework player controller / AI scripts](https://app.clickup.com/t/mb5m9k)

**Review by devildrake — CHANGES_REQUESTED** [review]
Found bugs:
- HUD ability cooldown effect not playing
- Most sounds not playing
- Shooting doesn't work properly (Maybe attackspeed or animation related issue)
- Onimaru doesn't have animations
- Fang animation on shooting doesn't play properly
- Dash has a wierd feel to it

File: Penteract/Assets/Scripts/Player.cpp [hunk 1]
```diff
@@ -1,6 +1,141 @@
...
+
+	if (Input::GetKeyCode(Input::KEYCODE::KEY_D)) {
+		if (md == MovementDirection::UP) md = MovementDirection::UP_RIGHT;
+		else if (md == MovementDirection::DOWN) md = MovementDirection::DOWN_RIGHT;
+		else md = MovementDirection::RIGHT;
+	}
+	return md;
+}
+
+int Player::GetMouseDirectionState() {
```
[the remark this answers is not in the document]
Comment (Yonimevicio) [reply Penteract/Assets/Scripts/Player.cpp:62]: Hmm okay it's a good idea

File: Penteract/Assets/Scripts/Player.cpp [hunk 2]
```diff
@@ -1,6 +1,141 @@
...
+#include "Geometry/Plane.h"
+#include "Geometry/Frustum.h"
+#include "Geometry/LineSegment.h"
 
 void Player::SetAttackSpeed(float attackSpeed_)
 {
 	attackSpeed = attackSpeed_;
 }
+
+void Player::Hit(int damage_)
```
[the remark this answers is not in the document]
Comment (Yonimevicio) [reply Penteract/Assets/Scripts/Player.cpp:11]: okay

File: Penteract/Assets/Scripts/CameraController.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include "Scripting/Script.h"
+
+class CameraController : public Script
```
[the remark this answers is not in the document]
Comment (Yonimevicio) [reply Penteract/Assets/Scripts/CameraController.h:5]: yes, all camera code will go in this class

**Review by Jordi-Gil — APPROVED** [review]

File: Penteract/Assets/Scripts/Player.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yonimevicio) [reply Penteract/Assets/Scripts/Player.cpp:62]: I will change it in another PR when I include the diagonal animations.

**Review by jcarlos0305 — DISMISSED** [review]

**Review by Jordi-Gil — DISMISSED** [review]

**Review by devildrake — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Pull request merged**